client/gui/UiMutex.h

```diff
@@ -0,0 +1,16 @@
+/*
+ * UiMutex.h, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+#pragma once
+
+#ifdef VCMI_HTML5_BUILD
+typedef boost::recursive_mutex ui_mutex;
+#else
+typedef boost::mutex ui_mutex;
+#endif
```

Comment:
Why different mutex is needed for web build?

Reply:
Good question. Probably in the first build, I didn’t have a separate thread for the server to run, and sometimes it happened that this mutex was entered multiple times from the main thread. However, in the current build, the server is executed in a separate thread, so this is probably not needed anymore. I will check this.